### ISSUE TITLE
Option to Turn off or hide YouTube shorts

### ISSUE THREAD

**Issue by sahlers83** [TeamNewPipe/NewPipe#7644]
### Checklist

- [x] I checked, but didn't find any duplicates (open OR closed) of this issue in the repo. 
- [x] I have read the contribution guidelines given at https://github.com/TeamNewPipe/NewPipe/blob/HEAD/.github/CONTRIBUTING.md.
- [x] This issue contains only one feature request. I will open one issue for every feature I want to request.

#### What feature do you want?

Add a filter in "Settings -> Content " with title "Include YouTube Shorts" with description "Turn off to hide YouTube shorts".  (Lets say one can place this option after Load Thumbnails.

#### Why do you want this feature?

1. Subscribe to a channel
2. Go to Hamburger menu -> Subscriptions -> Open the channel

Some channels are just having too many "shorts", and the user can avoid it.

Note that this feature is NOT connected to "Whats New" option. In the list of Whats New, YouTube shorts vides DO NOT appear

**Comment by SameenAhnaf** [comment]
I highly dislike Shorts and find disturbing when mixed with usual content. YouTube app shows Shorts both in row and other videos. However, Newpipe should exclusively show them in a row which could be optionally hidden just like clicking on arrow of YouTube video description.

**Comment by eladkarako** [comment]
`/shorts/` links will as of start of April 2022 get redirects (`HTTP 303 See Other`)  to back to `/watch?v=` normal video. you can still handle "short" videos (under 2 minutes).  

I'm applying <a href="https://gist.github.com/eladkarako/2549ac9cdfd7eae43a957404ec5e339b">a uBlock Origin rule</a> with `:upward` for DOM-traversing (avoiding APIs), 
but you shouldn't bother with that. You pretty much have all the video details already so you can introduce hiding to "short" video based on length (in seconds).

**Comment by Magneticdud** [comment]
A filter to hide shorts is becoming an essential feature 

More and more channels are having too many shorts, very irritating

**Comment by atul1503** [comment]
How to see shorts? Is there any exclusive button for shorts in NP?

**Comment by atul1503** [comment]
Can channel groups be configured to only show shorts from the channels which are part of it?

**Comment by Magneticdud** [comment]
At the moment it's not possible. The purpose of this proposal is to **hide** shorts from the channels, though, not to make them prominent

**Comment by terminal-root** [comment]
I don't know much about the specs behind shorts, but I imagine one way to do it would be to filter by aspect ratio.

Another good place to be able to hide shorts is the "What's New" feed. They take up a lot of space there.

**Comment by SwallowYourDreams** [comment]
I second all of the above. Users who find shorts a waste of bandwidth and brain cells should he able to filter them out of search results and the "What's new" feed.

**Comment by Eiion** [comment]
> In the list of Whats New, YouTube shorts vides DO NOT appear

Incorrect. Maybe back in January but nowadays YouTube Shorts indeed do appear on the What's New tab.

**Comment by Eiion** [comment]
"Subscription lists"? Seems like a way of overcomplicating things when a simple filter can just "hide" the things that you don't want to see.
I guess if you only want shorts you could do the same the other way around but filter everything that's not videos. But since New Pipe is a tool created to watch videos from multiple platforms in one, not short-form content, I'm not sure if that is something that even fits the description. Because then you could also argue why not also include Facebook/Instagram/snapchat/tiktok stories as well if we make shorts a separately available thing.

**Comment by newpiper** [comment]
I use the version 0.24.1 - lates for now. But I do not see any option about shorts on settings.

We may upload shorts and see the audiences reaction. Also some users that i talk about Np says mostly "we like shorts and want to watch it but this app has not include it" so we quit.

I think shorts must be in this app. But current there is no option for this.

**Comment by Magneticdud** [comment]
You're wrong. Newpipe unfortunately shows all the shorts crap. The option need to be added to **hide** them

**Comment by newpiper** [comment]
I just want to let you know that actually you are wrong. I am using 0.24.1 version, I subscribed my own channel. But when I click it, I do not see any **shorts video, I only see normal videos**

I read the all replies on this topic. But there is no option on "setting>content" menu - like **show/hide shorts videos**. Let me know if you have the answer, so I can check asap. I can also upload a screenshot if necessary.

**Comment by Magneticdud** [comment]
If that's the case that's perfect, the purpose of the issue is to **hide** that

I am not sure because when I unfollow all creators that post too many shorts

**Comment by torriem** [comment]
If you are using the option, "Fetch from dedicated feed when available," which is the default if you have many channels in your subscription, you get everything including shorts.   Until Youtube fixes the RSS feed system to separate shorts like the have in their normal interface, I don't see how FreeTube can filter. There's literally no way to distinguish shorts from normal videos in the RSS feed. The description has nothing in it unless the creator voluntarily puts in a hash tag or something.  I can only tell the shorts when the thumbnail is obviously vertical.  I cannot see lengths either (length is not included in the RSS feed).

**Comment by DJiK** [comment]
> > In the list of Whats New, YouTube shorts vides DO NOT appear
> 
> Incorrect. Maybe back in January but nowadays YouTube Shorts indeed do appear on the What's New tab.

For me, shorts appear nowhere. 🤷‍♂️
I must go on the YouTube site to see them in the channels.

**Comment by Eiion** [comment]
Maybe your subscribed channels don't upload many and you didn't notice. My what's new tab is full of shorts (especially since they got made monetizable recently) and I can hardly stand it anymore.

**Comment by DJiK** [comment]
> Maybe your subscribed channels don't upload many and you didn't notice. My what's new tab is full of shorts (especially since they got made monetizable recently) and I can hardly stand it anymore.

No. 
![Screenshot_20230227_143244_Opera](https://user-images.githubusercontent.com/66468470/221577481-7d6d26fc-586d-47d7-8e7f-4c784955474b.jpg)
![Screenshot_20230227_143215_NewPipe](https://user-images.githubusercontent.com/66468470/221577513-348fc0ec-95ba-400f-9a57-3739ede223db.jpg)

**Comment by Magneticdud** [comment]
For me i see shorts only in that cesspool called "trending"

Luckily in my "what's new" tab there are no shorts at all, even if i subscribed to some channels that do that kind of content (example: LTT)

**Comment by DJiK** [comment]
Ah, but in Trending there is everything, so much content. I only care about my subscriptions and their channels.
The one I take in example only do shorts, so I need to go to Youtube.com to see them which not the goal of using NewPipe. ;-)

**Comment by mirko77** [comment]
Shorts are hidden but they still appear during playback. If I pick a channel and play its videos in sequence, shorts are played.
